doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,625 @@
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+|          NTP             |  Network Time Protocol              |
+|          ntpd            |  NTP Daemon                         |
+| mgmt VRF                 | Management VRF                      |
+
+
+# 1 Feature Overview
+
+NTP stands for Network Time Protocol. It is used to synchronize the time of a computer or server to another server or reference time source.   
+
+SONiC click CLI provides commands to add and delete the IP address of a remote NTP server. Multiple NTP servers can be configured, and both IPv4 and IPv6 are supported. SONiC click CLI also provides the show command (output of "ntpq -pn").   
+
+This feature provides users the same capabilities via Management CLI, REST and gNMI using OpenConfig YANG models.
+In addition, it provides users to configure NTP source interface and NTP vrf. For this release, only "mgmt" and "default" vrf can be configured as NTP vrf.
+
+## 1.1 Requirements
+
+### 1.1.1 Front end configuration and get capabilities
+
+#### 1.1.1.1 add/delete NTP server      
+Add/delete NTP server information in the Redis ConfigDB and in the ntp.conf file (NTP configuration file).   
+The NTP server can be IPv4 address, IPv6 address , or a ntp pool name.   
+Mutliple NTP servers can be configured.     
+      
+#### 1.1.1.2 add/delete NTP source interface
+Add/delete the global NTP source interface in the Redis ConfigDB and in the ntp.conf file. The ip address of this interface will be used by ntpd as source ip for all NTP packets.     
+Only one NTP source interface can be configured. 
+Following interfaces can be used as NTP source interface when "mgmt" is not configured as NTP vrf.    
+- Ethernet interface    
+- PortChannel    
+- Vlan interface    
+- Loopback interfacee    
+- eth0    
+      
+#### 1.1.1.3 add/delete VRF name
+Add/delete the global NTP VRF information in the Redis ConfigDB. It is used by /etc/init.d/ntp script to start the ntpd in a specific VRF context.
+For this release, only Management VRF and default instance are supported.         
+        
+#### 1.1.1.4 Get NTP association
+This displays the output of "ntpq -np" command.  
+    
+#### 1.1.1.5 Overall Behavior related to NTP source interface and NTP vrf     
+##### When mgmt VRF is configured      
+a.if no ntp vrf is configured, ntp service starts in mgmt VRF context by default           
+b.if "mgmt" is configured as NTP vrf, ntp service starts in mgmt VRF context        
+c.if "default" is configured as NTP vrf, ntp service starts in default vrf context      
+     
+##### When mgmt VRF is not configured     
+ntp service always starts in default vrf context      
+      
+##### NTP source interface related      
+a.if NTP vrf is empty or "default", NTP source interface can be configured     
+b.if NTP vrf is "mgmt", configuration of “ntp source-interface” will be rejected     

```

Comment:
@bsun-sudo Why is this restriction there? It appears that Vlan SVI's can be set to be "ip vrf forwarding mgmt" i.e. Management VRF. We are getting requests from the field on the same.

Reply:
This restriction is removed in the latest update.